rules/no-reduce.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+const methodSelector = require('./utils/method-selector');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const message = 'Array.reduce not allowed';
+
+const PROTOTYPE_SELECTOR = [
+	methodSelector({names: ['call', 'apply']}),
+	'[callee.object.type="MemberExpression"]',
+	':matches([callee.object.object.type="ArrayExpression"], [callee.object.object.object.type="Identifier"][callee.object.object.object.name="Array"][callee.object.object.property.name="prototype"])',
```

Comment:
ArrayExpression  add `elements.length=0`

Ignore non-empty arrays

Reply:
You also forgot `[callee.object.property.name="reduce"]`

Reply:
It's present on line#13 below.

Reply:
Right, missed.